.github/workflows/upgrade-environment.yml

```diff
@@ -0,0 +1,230 @@
+name: Test Upgrade Environment
+run-name: Creating ${{ github.event.inputs.deployment_name }} by @${{ github.actor }}
+
+on:
+  # Ability to execute on demand
+  workflow_dispatch:
+    inputs:
+      deployment_name:
+        type: string
+        description: |
+          Name with letters, numbers, hyphens; start with a letter. Max 20 chars. e.g., 'my-env-123'
+        required: true
+      elk-stack-version:
+        required: true
+        description: "Stack version: For released/BC version use 8.x.y, for SNAPSHOT use 8.x.y-SNAPSHOT"
+        default: "8.11.0"
+        type: string
+      docker-image-override:
+        required: false
+        description: "Provide the full Docker image path to override the default image (e.g. for testing BC/SNAPSHOT)"
+
+env:
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  AWS_REGION: "eu-west-1"
+  WORKING_DIR: deploy/test-environments
+  FLEET_API_DIR: fleet_api/src
+  TF_VAR_stack_version: ${{ inputs.elk-stack-version }}
+  TF_VAR_ess_region: gcp-us-west2
+  TF_VAR_ec_api_key: ${{ secrets.EC_API_KEY }}
+  DOCKER_IMAGE: ${{ inputs.docker-image-override }}
+
+jobs:
+  init:
+    runs-on: ubuntu-20.04
+    outputs:
+      stack-version: ${{ steps.set-previous-version.outputs.PREVIOUS_VERSION }}
+      ess-region: ${{ env.TF_VAR_ess_region }}
+    steps:
+      - name: Set Previous Version
```

Comment:
Should be a separate script both for readability and for reusability. Do you think we can use APM [pipeline](https://github.com/elastic/apm-pipeline-library/blob/9c712c2558ef07309c4ddf3cf02d20a1cb2b1b3a/updatecli/updatecli.d/bump-aliases.yml#L43C1-L55C31) instead?

Reply:
Since the init job currently consists of raw commands, creating a script in the repository would necessitate an additional action for checkout in order to retrieve this script. As a result, I've initiated a [task](https://github.com/elastic/cloudbeat/issues/1522) to refactor the init job. This refactoring will include an investigation into whether the APM pipeline can be reused instead of creating our own script.

Reply:
I think the script should be moved into a separate file, either using the APM pipeline or placing it in the scripts folder. Adding checkout step should be pretty straight forward

Reply:
Added checkout step and script.